torch/csrc/jit/passes/quantization.cpp

```diff
@@ -828,5 +828,44 @@ graph(%self, %x):
   }
 }
 
+void FoldQuantize(
+    script::Module& module,
+    const std::string& method_name) {
+  const std::string pattern = R"(
+graph(%self, %scale, %zero_point, %dtype):
+   %weight = prim::GetAttr[name="weight"](%self)
+   %weight_quant = aten::quantize_linear(%weight, %scale, %zero_point, %dtype)
+   return (%weight_quant))";
+  Graph pattern_graph;
+  std::unordered_map<std::string, Value*> vmap;
+  script::parseIR(pattern, &pattern_graph, vmap);
+  auto method = module.get_method(method_name);
+  auto graph = method.graph();
+  auto matches = findPatternMatches(pattern_graph, *graph);
+  for (const auto& match: matches) {
+    auto match_vmap = match.values_map;
+    auto* weight = match_vmap.at(vmap.at("weight"));
+    TORCH_INTERNAL_ASSERT(
+        weight->node()->kind() == prim::GetAttr,
+        "Expected to get weight from prim::GetAttr");
+    TORCH_INTERNAL_ASSERT(
+        weight->node()->s(attr::name) == "weight",
+        "Expected attribute name to be weight but got: ", weight->node()->s(attr::name));
+    auto float_weight = module.get_parameter("weight").variable_data();
+    auto scale = toIValue(match_vmap.at(vmap.at("scale"))).value().toDouble();
```

Comment:
If some module happens to have "weight" but doesn't have "scale" this function will fail (imagine e.g. there is a conv that we're not quantizing). We should bail out gracefully in such cases.